src/test/sitl/sitl_harness.py

```
@@ -0,0 +1,440 @@
+            except OSError:
+                return
+        raise RuntimeError("previous SITL still holds the MSP port")
+
+    def start(self):
+        # The TCP listener has no SO_REUSEADDR; sockets from a previous scenario
+        # lingering in TIME_WAIT can make the bind fail silently, so retry the
+        # whole launch a few times rather than only polling for the port.
+        for attempt in range(3):
+            self.wait_port_free()
+            logf = open(os.path.join(self.workdir, "sitl.log"), "a")
+            self.proc = subprocess.Popen([self.binary], cwd=self.workdir, stdout=logf, stderr=logf)
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)